src/Kestrel.Core/Internal/Http/Frame.cs

```diff
@@ -345,11 +345,11 @@ public void Reset()
             StatusCode = StatusCodes.Status200OK;
             ReasonPhrase = null;
 
-            RemoteIpAddress = RemoteEndPoint?.Address;
-            RemotePort = RemoteEndPoint?.Port ?? 0;
+            RemoteIpAddress = HttpConnectionFeature?.RemoteIpAddress;
+            RemotePort = HttpConnectionFeature?.RemotePort ?? 0;
```

Comment:
Also `HttpConnectionFeature` is a feature lookup property itself; so this is even worse... need a function local cache e.g.
```csharp
var httpConnectionFeature = HttpConnectionFeature;
// ... Do the stuff 
```
Else you are evaluating `ConnectionInformation.Features.Get<IHttpConnectionFeature>()` also 4x